skimage/restoration/tests/test_inpaint.py

```diff
@@ -6,14 +6,16 @@
 from skimage._shared.testing import assert_allclose
 
 
-def test_inpaint_biharmonic_2d():
-    img = np.tile(np.square(np.linspace(0, 1, 5)), (5, 1))
+@testing.parametrize('dtype', [np.float32, np.float64])
+def test_inpaint_biharmonic_2d(dtype):
+    img = np.tile(np.square(np.linspace(0, 1, 5, dtype=dtype)), (5, 1))
     mask = np.zeros_like(img)
     mask[2, 2:] = 1
     mask[1, 3:] = 1
     mask[0, 4:] = 1
     img[np.where(mask)] = 0
     out = inpaint.inpaint_biharmonic(img, mask)
+    assert out.dtype == img.dtype
```

Comment:
This will work for the current tests but would fail if we add say `np.float16` to line 9. That is if I understand the current idea correctly to return the image in the same type it was processed in.

Reply:
yes, in that case we have to change to `assert out.dtype == _float_type(img.dtype)`